src/__tests__/index.test.js

```diff
@@ -1,30 +1,56 @@
 import React from 'react';
 import { Text } from 'react-native';
-import { create } from 'react-test-renderer';
+import { render } from '@testing-library/react-native';
 
 import HighlightText from '../';
 
 describe('<HighlightText />', () => {
-  it('should render properly', () => {
-    const wrapper = create(
-      <HighlightText
-        searchWords={['hello', 'world']}
-        textToHighlight={'In this text we can do a hello to the entire world'}
-      />
-    );
-    expect(wrapper.toJSON()).toMatchSnapshot();
+  it('should wrap each matched search word with an isolated element', () => {
+    const props = {
+      searchWords: ['hello', 'world'],
+      textToHighlight: 'In this text we can do a hello to the entire world',
+    };
+    const { queryByText, toJSON } = render(<HighlightText {...props} />);
+
+    for (const word of props.searchWords) {
+      expect(queryByText(word)).toBeTruthy();
+    }
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should detect all ocurrencies of searched words in textToHighlight prop', () => {
+    const props = {
+      searchWords: ['Slim', 'Shady'],
+      textToHighlight:
+        "I'm Slim Shady, yes, I'm the real Shady. All you other Slim Shadys, are just imitating. So won't the real Slim Shady, please stand up?",
+    };
+    const { queryAllByText, toJSON } = render(<HighlightText {...props} />);
+
+    const allSlims = queryAllByText('Slim');
+    const allShadys = queryAllByText('Shady');
+
+    expect(allSlims).toHaveLength(3);
+    expect(allShadys).toHaveLength(4);
```

Comment:
Like the previous comment, you should use `getBy` here and these variables aren't needed 😉 
```suggestion
    expect(getAllByText('Slim')).toHaveLength(3);
    expect(getAllByText('Shady')).toHaveLength(4);
```

Reply:
In this case, it does make sense to use get instead of query, since we're checking the content right after. About the variables... Ok, it makes sense, but I don't think it makes any difference haha The compiler will still need to save the same amount of memory to store the values. Anyway, those are good catches 😄

Reply:
Now it is broken 🤣   

Just kidding, Sonar saw our changes as code smells haha

Reply:
Found the error, you forgot to declare `getAllByText` method (stop trolling my hacktoberfest man 😝  )

Thank you for the help =D